subprojects/dependency-management/src/main/java/org/gradle/internal/resolve/ModuleVersionResolveException.java

```diff
@@ -130,10 +131,15 @@ private String toString(ComponentIdentifier identifier) {
 
     protected ModuleVersionResolveException createCopy() {
         try {
-            return getClass().getConstructor(ComponentSelector.class, String.class).newInstance(selector, getMessage());
+            String message = getMessage();
```

Comment:
FTR this is done eagerly because I assume this `createCopy` code is called to detach state. If we used the `Factory` we would have shared state between the different copies.